### PR TITLE
RHEL8 spec update to require moby-engine or docker-ce

### PR DESCRIPTION
**Pull request by brotherneeru**
Adding a requires clause to spec for rhel8 "Requires: (moby-engine or docker-ce)", as well as moving check for docker group to %post.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ X] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ X] Title of the pull request is clear and informative.
- [X ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: edgelet/contrib/enterprise-linux/aziot-edge.spec [hunk 1]
```diff
@@ -62,19 +63,6 @@ make \
...
-    echo " ERROR: No container runtime detected."
-    echo ""
-    echo " Please install a container runtime and run this install again."
-    echo ""
-    echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
-
-    exit 1
-fi
-
 # Create iotedge group
```
Comment (arsing) [inline edgelet/contrib/enterprise-linux/aziot-edge.spec:66]: You need to move all of this to `%post`. What you have right now won't work.

Please test it to make sure it works.

**Comment by arsing** [comment]
What about `edgelet/contrib/centos/aziot-edge.spec` ? Did you test that it does not support `or` ?

**Comment by brotherneeru** [comment]
> What about `edgelet/contrib/centos/aziot-edge.spec` ? Did you test that it does not support `or` ?

Just tested this. Does not support or

**Comment by arsing** [comment]
Cool, so just tese these scenarios on EL8 to be sure:

1. With neither Docker repo nor MS repo added, `yum install aziot-edge` should fail because it can't satisfy the requirement.

1. With just Docker repo added but not MS repo, `yum install aziot-edge` should automatically pull in `docker-ce`.

1. With just MS repo added but not Docker repo, `yum install aziot-edge` should automatically pull in `moby-engine`.

1. With both Docker and MS repos added, what does it do?

**Comment by brotherneeru** [comment]
> Cool, so just test these scenarios on EL8 to be sure:
> 
> 1. With neither Docker repo nor MS repo added, `yum install aziot-edge` should fail because it can't satisfy the requirement.
> 2. With just Docker repo added but not MS repo, `yum install aziot-edge` should automatically pull in `docker-ce`.
> 3. With just MS repo added but not Docker repo, `yum install aziot-edge` should automatically pull in `moby-engine`.
> 4. With both Docker and MS repos added, what does it do?

Defaults to docker repo

**Comment by arsing** [comment]
Hmm, it would be nice if it preferred `moby-engine`, but I guess it's fine.

**Review by arsing — APPROVED** [review]

**Review by yophilav — APPROVED** [review]

**Pull request merged**